flink-yarn/src/main/scala/org/apache/flink/yarn/YarnJobManager.scala

```diff
@@ -155,12 +156,14 @@ class YarnJobManager(
 
       rmClientOption foreach {
         rmClient =>
-          Try(rmClient.unregisterApplicationMaster(status, diag, "")).recover{
+          Try(rmClient.unregisterApplicationMaster(status, diag, "")).recover {
+            // https://issues.apache.org/jira/browse/YARN-1842
+            case e : InvalidApplicationMasterRequestException =>
```

Comment:
How about at least logging the exception to make sure we are not silently swallowing the exception.